### PR TITLE
Add low-fidelity CLI to schedule tests

### LINKED ISSUE #68: CLI should be able to make RPC calls to the service

### LINKED ISSUE #69: CLI proxies results to the client

### PR DESCRIPTION
**Pull request by codeblooded**
This change adds a CLI to interact with the service, scheduling tests and reporting their progress. This CLI is not the final version that will be released, so it is lacking in tests and aesthetics.

### CODE REVIEW

**Comment by codeblooded** [comment]
This low-fidelity CLI does not support all the features we intend, such as reading scenarios from a file. I have opened #97 to track this enhancement for a future PR.

File: testctrl/cmd/testctl/main.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+func exit(code int, messageFmt string, args ...interface{}) {
+	fmt.Fprintf(os.Stderr, messageFmt+"\n", args...)
+	os.Exit(code)
+}
+
+// Schedule accepts command line arguments and uses them to schedule a test,
+// reporting progress as it runs.
+func Schedule(args []string) {
+	flags := ScheduleFlags{}
+	scheduleFlags := flag.NewFlagSet("testctl schedule", flag.ExitOnError)
```
Comment (codeblooded) [inline testctrl/cmd/testctl/main.go]:
```suggestion
	scheduleFlags := flag.NewFlagSet("testctl", flag.ExitOnError)
```

File: testctrl/cmd/testctl/main.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"reflect"
+	"time"
+
+	grpcPb "github.com/codeblooded/grpc-proto/genproto/grpc/testing"
```
Comment (codeblooded) [inline testctrl/cmd/testctl/main.go]:
```suggestion
	grpcpb "github.com/codeblooded/grpc-proto/genproto/grpc/testing"
```

File: testctrl/cmd/testctl/main.go [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+// Set parses a client flag and appends it to the list.
+func (cl *clientList) Set(client string) error {
+	cl.clients = append(cl.clients, client)
+	return nil
+}
+
+// scenario wraps the scenario protobuf, implementing the flag.Value interface.
+// This allows it to be parsed alongside flags with primitive types.
+type scenario struct {
+	proto *grpcPb.Scenario
```
Comment (codeblooded) [inline testctrl/cmd/testctl/main.go]:
```suggestion
	proto *grpcpb.Scenario
```

File: testctrl/cmd/testctl/main.go [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+}
+
+// Set parses the JSON string into a protobuf as the flag is parsed. It returns
+// an error is the flag is malformed or cannot be marshaled into a proto.
+func (sc *scenario) Set(scenarioJSON string) error {
+	if scenarioJSON == "" {
+		return errors.New("a valid scenario is required, but missing")
+	}
+
+	sc.proto = &grpcPb.Scenario{}
```
Comment (codeblooded) [inline testctrl/cmd/testctl/main.go]:
```suggestion
	sc.proto = &grpcpb.Scenario{}
```

**Review by paulosjca — APPROVED** [review]

**Pull request merged**